### PR TITLE
fixes #9410 feat(nimbus): Fetch Firefox Desktop manifests with manifest-tool

### LINKED ISSUE #9410: Vendor m-c experimenter yaml via manifesttool
Desktop feature manifest versioning is done by a bash script in the Makefile. We should port this to the manfiesttool added in [https://mozilla-hub.atlassian.net/browse/EXP-3836](https://mozilla-hub.atlassian.net/browse/EXP-3836|smart-link).

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-3837)

### PR DESCRIPTION
**Pull request by brennie**
Because

- we want to support fetching mozilla-central with manifest-tool

This commit

- adds support for hg.mozilla.org repositories to manifest-tool;
- adds support for legacy experimenter.yaml-based apps to manifest-tool;
- adds Firefox Desktop to apps.yaml; and
- updates make feature_manifests to fetch Firefox Desktop manifests.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -112,42 +112,42 @@ services:
...
-    environment:
-      - CIRRUS=http://cirrus:8001
+  # cirrus:
+  #   build:
+  #     context: cirrus/server/
+  #     dockerfile: Dockerfile
+  #   env_file: .env
+  #   links:
+  #     - kinto
+  #   volumes:
```
Comment (yashikakhurana) [inline docker-compose.yml:122]: I see cirrus stuff commented out, is there a reason?

Comment (brennie) [reply docker-compose.yml]: accidental commit. it prevents building cirrus, which without hydrobuild was taking a long time

File: experimenter/manifesttool/hgmo_api.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    file_path: str,
+    rev: str,
+    download_path: Path,
+):
+    url = f"{HGMO_URL}/{repo}/raw-file/{rev}/{file_path}"
+    with requests.get(url, stream=True) as req:
+        req.raise_for_status()
+        with download_path.open("wb") as f:
+            for chunk in req.iter_content(chunk_size=8192):
+                f.write(chunk)
```
Comment (jaredlockhart) [inline experimenter/manifesttool/hgmo_api.py:37]: Why do we have to handle it as a stream? 🤔

**Review by jaredlockhart — APPROVED** [review]
There's a lot going on in here, could've been split up to just add the hg stuff, then the change from internal schema to external schema, etc.  But it all looks good to me thnx @brennie 🎉 🎉 🎉 🎉 🎉 🎉

File: experimenter/manifesttool/hgmo_api.py (hunk 2 above)
Comment (brennie) [reply experimenter/manifesttool/hgmo_api.py:37]: We don't *have* to, just best practices for downloading files. It wasn't much harder to write than read the entire response and write the entire file.

File: experimenter/experimenter/features/__init__.py [hunk 3]
```diff
@@ -31,41 +30,24 @@ class FeatureVariableType(Enum):
...
-    description: Optional[str]
-    exposureDescription: Optional[Union[Literal[False], str]]
-    isEarlyStartup: Optional[bool]
+@dataclass
+class Feature:
     slug: str
-    variables: Optional[dict[str, FeatureVariable]]
-    schema_paths: Optional[FeatureSchema] = Field(alias="schema")
+    application_slug: str
+    model: Union[FeatureWithExposure, FeatureWithoutExposure]
```
[the remark this answers is not in the document]
Comment (brennie) [reply experimenter/experimenter/features/__init__.py:36]: So there is a top level `Feature` type, but it is

```
class Feature(BaseModel):
    __root__: Union[FeatureWithExposure, FeatureWithoutExposure]
```

which means to access its properties you need to do `feature.__root__.property` which is awkward. I couldn't think up a better name to wrap the model Union vs the class so I left it as is.

[the remark this answers is not in the document]
Comment (jaredlockhart) [reply experimenter/experimenter/features/__init__.py:36]: Oh hrmmm yeah.  I wonder if there's a better pattern for this.

**Pull request merged**